src/Hl7.Fhir.Specification/Specification/Source/Summary/ArtifactSummaryDetailsCollection.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Hl7.Fhir.Specification.Source.Summary
+{
+    /// <summary>Common interface for retrieving artifact summary details by key.</summary>
+    /// <remarks>
+    /// Implemented by <see cref="ArtifactSummaryDetailsCollection"/> as well as <see cref="ArtifactSummary"/>.
+    /// Allows common extension methods for retrieving specific summary details.
+    /// </remarks>
+    public interface IArtifactSummaryDetailsProvider
```

Comment:
Can we not combine ArtifactSummary and ArtifactSummaryCollection?  Why would we keep both?

Reply:
I decided to encapsulate the backing store. Client code just needs an indexer property this[key], so this is the interface expresses. Implementation is free to choose a specific back store, we could change/optimize this at a later time.
However IReadOnlyDictionary seems to fit this use case, so we could decide to use this and drop the custom interface. At the cost of a bit more work when we'd want to change the backing store (to implement the additional methods).

I want to prevent consumers from changing harvested properties. However I also don't want to repeat all this property logic. So I decided on a kind of builder pattern. The extractor fills a read/write collection, then hands it over to initialize a read-only summary instance, which saves and wraps the r/w collection. By implementing a common interface, we can re-use extension methods on both the r/w collection class and the final r/o summary instance.